backend/dataall/modules/datapipelines/db/datapipelines_repositories.py

```diff
@@ -25,11 +32,52 @@ def count_resources(self, session, environment, group_uri) -> int:
         )
 
     @staticmethod
-    def get_clone_url_http(session, environmentUri, repo):
-        env: Environment = session.query(Environment).get(
-            environmentUri
+    def create_pipeline(
+            session,
+            username: str,
+            admin_group: str,
+            uri: str,
+            data: dict = None,
+    ) -> DataPipeline:
+        environment = EnvironmentService.get_environment_by_uri(session, uri)
+
+        pipeline: DataPipeline = DataPipeline(
```

Comment:
I think instantiation of the object should happen in the service

Reply:
I was also doubtful on this one because I found examples of us creating the RDS object in the services and in db. The logic I followed was: if I am a developer and I want to see how a DataPipeline was created in the database I will go to the db package. But I am not completely convinced by one or other approach. Why do you think it should be in `services`?

Reply:
I think the mixed creation of objects in services and db happens due to differences around the object creation (eg. in some cases multiple things are done besides just creating the object). In this case, besides creating the object nothing else really happens, so I agree that its nicer to have it in db - I just remembered the pattern that object creation was done in services in many places.
Best would be to have all the object creations in either db or services, but its fine from my side to do this in a followup, also after the release.